Add 26 Missing GitLab Settings

Jitsusama opened the pull request:
I have discovered ~ 60 settings that have been added to GitLab's API but are currently missing in this project. In this PR I'm taking the next stab at correcting this by adding in 26 missing settings, leaving 27 left.

These settings were found by perusing documentation and code in response to [gitlab-org/terraform-provider-gitlab#6332](https://gitlab.com/gitlab-org/terraform-provider-gitlab/-/issues/6332).

svanharmelen commented on types.go:
```diff
@@ -230,6 +230,17 @@ func Availability(v AvailabilityValue) *AvailabilityValue {
+	AllowForcePush          bool  `json:"allow_force_push,omitempty"`
+	AllowedToMerge          []int `json:"allowed_to_merge,omitempty"`
+	DeveloperCanInitialPush bool  `json:"developer_can_initial_push,omitempty"`
+}
```
I prefer to have this struct defined below the `Settings` struct.

Jitsusama replied:
This should be addressed now. Let me know if you would like it moved somewhere differently, I assumed you meant the bottom of the file, but I just recognized that you might have meant right below. I don't mind moving it to wherever you'd prefer.

svanharmelen commented on settings.go:
```diff
@@ -462,359 +463,360 @@ func (s *SettingsService) GetSettings(options ...RequestOptionFunc) (*Settings,
+	DefaultArtifactsExpireIn                              *string                   `url:"default_artifacts_expire_in,omitempty" json:"default_artifacts_expire_in,omitempty"`
+	DefaultBranchName                                     *string                   `url:"default_branch_name,omitempty" json:"default_branch_name,omitempty"`
+	DefaultBranchProtection                               *int                      `url:"default_branch_protection,omitempty" json:"default_branch_protection,omitempty"`
+	DefaultBranchProtectionDefaults                       *BranchProtectionDefaults `url:"default_branch_protection_defaults,omitempty" json:"default_branch_protection_defaults,omitempty"`
```
This field should be set to a `BranchProtectionDefaultsOptions` struct which contains the required fields as pointers (see other option structs).

Jitsusama replied:
This should be addressed now.

svanharmelen reviewed (changes requested):
Looks good, but one comment related to the additional struct.

Jitsusama commented:
@svanharmelen; mind taking another look at this, I think I have addressed all of your concerns and would love to cut another MR soon with the last batch of changes.

timofurrer commented:
@Jitsusama while you are at it, do you mind adding `receptive_cluster_agents_enabled` that was recently added with https://gitlab.com/gitlab-org/gitlab/-/merge_requests/162514 ?

Jitsusama commented:
> @Jitsusama while you are at it, do you mind adding `receptive_cluster_agents_enabled` that was recently added with https://gitlab.com/gitlab-org/gitlab/-/merge_requests/162514 ?

I'll add it to the next batch that covers the Rs @timofurrer!

svanharmelen reviewed (approved):
I updated the ordering (minor tweak), but looks good now 👍🏻  Thanks!
